### PR TITLE
Move config to configuration

### PR DESCRIPTION
**Pull request by jenshnielsen**
This resolves the ambiguity noted in #1645 by moving the config module to configuration such that

* qcodes.config  is the current active config
* qcodes.configuration is the configuration module.

- [x] Update any notebooks and documentation. 
- [x] I have chosen not to leave around the old alias as it was super confusing and not clear how that would resolve compared to the current config. 

@QCoDeS/core This is now ready for final review

### CODE REVIEW

**Comment by WilliamHPNielsen** [comment]
@jenshnielsen, to answer your question
> Should we leave around qcodes.config as an alias for the time being and remove it later?

Wouldn't it work to leave it around with a deprecation warning like we've done for some of the `dataset` modules, e.g. `qcodes.dataset.sqlite_base.py`? If that's possible, I'd vote for us taking the same approach here.

**Comment by jenshnielsen** [comment]
@QCoDeS/core please have a look at the current state. This now incorporates @astafan8 s work on types from #1824

Before merging I need to:

* write a deprecation message for the changelog 
* double check if the docs needs updates

File: docs/user/configuration.rst [hunk 1]
```diff
@@ -25,49 +25,52 @@ Default config
...
 The following  table explains the default configuration of qcodes.
 
-.. jsonschema:: ../../qcodes/config/qcodesrc_schema.json
+.. jsonschema:: ../../qcodes/configuration/qcodesrc_schema.json
 
 
 Using the config
 ----------------
-QCoDeS exposes the default values in the module namespace as `defaults`
+QCoDeS exposes the default values in the module namespace as `config`
```
Comment (astafan8) [inline docs/user/configuration.rst]:
```suggestion
QCoDeS exposes the configuration values in the `qcodes` module namespace as `config`:
```

File: qcodes/__init__.py [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
 
-from qcodes.config import Config
+import qcodes.configuration as qcconfig
 from qcodes.logger import start_all_logging
 from qcodes.logger.logger import conditionally_start_all_logging
 from qcodes.utils.helpers import add_to_spyder_UMR_excludelist
 from .version import __version__
 
-config: Config = Config()
+config: qcconfig.Config = qcconfig.Config()
```
Comment (astafan8) [inline qcodes/__init__.py]: is the type annotation still needed here? :(
```suggestion
config = qcconfig.Config()
```

File: qcodes/tests/test_config.py [hunk 3]
```diff
@@ -152,39 +153,39 @@ def default_config(user_config: Optional[str] = None):
...
     """
-    home_file_name = qcodes.Config.home_file_name
-    schema_home_file_name = qcodes.Config.schema_home_file_name
-    env_file_name = qcodes.Config.env_file_name
-    schema_env_file_name = qcodes.Config.schema_env_file_name
-    cwd_file_name = qcodes.Config.cwd_file_name
-    schema_cwd_file_name = qcodes.Config.schema_cwd_file_name
-
-    qcodes.Config.home_file_name = ''
+    home_file_name = qcodes.configuration.Config.home_file_name
```
Comment (astafan8) [inline qcodes/tests/test_config.py]: is `qcodes.configuration.` prefix needed if `Config` class has alrady been imported into this namespace?

File: qcodes/__init__.py (hunk 2 above)
Comment (jenshnielsen) [reply qcodes/__init__.py]: yes

Comment (astafan8) [reply qcodes/__init__.py]: ok, I lack some mypy understanding

Comment (jenshnielsen) [reply qcodes/__init__.py]: I honestly don't know why but mypy fails without it

**Pull request merged**